### PR TITLE
ANDROID-11369 snackbar allow 10 seconds

### PR DESCRIPTION
**Pull request by jdelga**
### :goal_net: What's the goal?
Allow snackbar of 5 or 10 seconds if it does not have an action

### :construction: How do we do it?
* Add component property
* Add catalog option

### ☑️ Checks
- [X] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [X] Tested with dark mode.
- [X] Tested with API 21.

### :test_tube: How can I test this?
It can be tested in the dialog app

[mistica_10_result.webm](https://user-images.githubusercontent.com/36664452/203350341-802e3c61-a674-46e1-9229-6fc3bccc295f.webm)

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/fragment/SnackBarCatalogFragment.kt [hunk 1]
```diff
@@ -48,12 +51,16 @@ class SnackBarCatalogFragment : Fragment() {
             SnackbarBuilder(view, inputText.text.toString()).apply {
                 inputAction.text.toString().let { actionText ->
                     if (actionText.isNotEmpty()) {
-                        withAction(actionText, View.OnClickListener { })
+                        withAction(actionText, { })
                     }
                 }
+                val duration = when {
+                    snackbarLenght10.isChecked -> SnackbarLenght.LENGTH_10
+                    else -> SnackbarLenght.LENGTH_5
```
Comment (gmerinojimenez) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/fragment/SnackBarCatalogFragment.kt:59]: minor, but I would use SHORT and LONG, similar to for example how Toast does, so the name is decoupled from the duration (if in the future we want that SHORT takes 6 seconds we will not need to replace all uses)

**Review by gmerinojimenez — APPROVED** [review]
So fast! thanks!

File: library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt [hunk 2]
```diff
@@ -111,9 +110,23 @@ open class SnackbarBuilder(view: View?, text: String) {
         action.isAllCaps = false
     }
 
+    companion object {
+        private const val MAX_TEXT_LINES = 4
+    }
+}
+
+enum class SnackbarLenght {
```
Comment (dagonco) [inline library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt:118]: Got a type there!

File: library/src/main/java/com/telefonica/mistica/feedback/README.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ Builder allows SnackBar customization:
 * `withCallback(Callback callback)`
   * Adds a callback for dismiss action. Dismiss action by definition will only work when using a coordinator layout as anchor view for the SnackBar.
 
-Finally, depending on the type of SnackBar, use one of the following to display it:
+Finally, depending on the type of SnackBar, use one of the following to display it. These methods allow an argument to choose the duration of 5 second or 10.
+5 is the default.
 * `showInformative()`
+* `showInformative(SnackbarLenght.LENGTH_10)`
```
Comment (dagonco) [inline library/src/main/java/com/telefonica/mistica/feedback/README.md:27]: According the code, the valid values are SHORT and LONG. Perhaps I would detail both in the documentation.

File: library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt (hunk 2 above)
Comment (jdelga) [reply library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt]: Thanks! I introduced it all over the place!

File: library/src/main/java/com/telefonica/mistica/feedback/README.md (hunk 3 above)
Comment (jdelga) [reply library/src/main/java/com/telefonica/mistica/feedback/README.md]: Done

**Pull request merged**